suite/suite.go

```diff
@@ -138,8 +138,10 @@ func Run(t *testing.T, suite TestingSuite) {
 				suite.SetT(t)
 				defer failOnPanic(t)
 				defer func() {
+					r := recover()
```

Comment:
Wouldn't this break the `recover()` in `failOnPanic(t)` that gets called later?

IIRC `defer`s are stored in a stack, so this function would be executed first and executing this `recover()` without 're-`panic`ing' would break the subsequent, `failOnPanic` function?

Reply:
This is a good remark.

First, `failOnPanic(t)` is a no-op if it is called in a context where it does not panic.

Then, you are right: in case a test panics, this function takes precedence over the `failOnPanic(t)` L139 and `failOnPanic(t)` will be a no-op. But I added the body of `failOnpanic(t)` L157:L162, so this way, we don't change the current behavior:

```golang
t.Errorf("test panicked: %v\n%s", r, debug.Stack())
t.FailNow()
```

We can define a subfunction for those two lines if needed.

Note that the `failOnPanic(t)` is still relevant because the current function may panic. That is because it may run user code via `AfterTest()` or `TearDownTestSuite()`

Reply:
Would it make sense to refactor said `failOnPanic()` body (L159 - L160) out into a common function that can then be called from here and `failOnPanic()`?

Reply:
Good point. I renamed `failOnPanic(t)` to `recoverAndFailOnPanic(t)` and called the common function `failOnPanic(t, r)`